compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -905,17 +907,61 @@ public void visit(BLangArrayType arrayTypeNode) {
             BArrayType arrType;
             if (arrayTypeNode.sizes.length == 0) {
                 arrType = new BArrayType(resultType, arrayTypeSymbol);
+                arrayTypeSymbol.type = arrType;
+                markParameterizedType(arrType, arrType.eType);
+                resultType = arrayTypeSymbol.type;
             } else {
-                int size = arrayTypeNode.sizes[i];
-                arrType = (size == UNSEALED_ARRAY_INDICATOR) ?
-                        new BArrayType(resultType, arrayTypeSymbol, size, BArrayState.UNSEALED) :
-                        (size == OPEN_SEALED_ARRAY_INDICATOR) ?
-                                new BArrayType(resultType, arrayTypeSymbol, size, BArrayState.OPEN_SEALED) :
-                                new BArrayType(resultType, arrayTypeSymbol, size, BArrayState.CLOSED_SEALED);
-            }
-            resultType = arrayTypeSymbol.type = arrType;
+                BLangExpression size = arrayTypeNode.sizes[i];
+                if (size instanceof BLangLiteral) {
+                    arrType = ((Integer) (((BLangLiteral) size).getValue()) == UNSEALED_ARRAY_INDICATOR) ?
+                            new BArrayType(resultType, arrayTypeSymbol,
+                                    (Integer) (((BLangLiteral) size).getValue()),
+                                    BArrayState.UNSEALED) :
+                            ((Integer) (((BLangLiteral) size).getValue()) == OPEN_SEALED_ARRAY_INDICATOR) ?
+                                    new BArrayType(resultType, arrayTypeSymbol,
+                                            (Integer) (((BLangLiteral) size).getValue()),
+                                            BArrayState.OPEN_SEALED) :
+                                    new BArrayType(resultType, arrayTypeSymbol,
+                                            (Integer) (((BLangLiteral) size).getValue()),
+                                            BArrayState.CLOSED_SEALED);
+                    arrayTypeSymbol.type = arrType;
+                    markParameterizedType(arrType, arrType.eType);
+                    resultType = arrayTypeSymbol.type;
 
-            markParameterizedType(arrType, arrType.eType);
+                } else {
+                    Name pkgAlias = names.fromIdNode(((BLangSimpleVarRef) size).pkgAlias);
+                    Name typeName = names.fromIdNode(((BLangSimpleVarRef) size).variableName);
+
+                    BSymbol sizeSymbol = lookupMainSpaceSymbolInPackage(size.pos, env, pkgAlias, typeName);
+
+                    if (!(symTable.notFoundSymbol.equals(sizeSymbol))) {
```

Comment:
The` visit(BLangArrayType arrayTypeNode)` function does not return anything, so every `if` will get evaluated if I do it this way. So won't having the `else` block be a better approach?

Reply:
You can just do `return;`.

Reply:
Or in this case, because it is within a for loop, and we may want to analyze the rest of the dimensions even if one results in an error, we can do a `continue;` maybe. I guess we'll have to maintain if at least one dimension was erroneous to set the final resultant type.

Reply:
So just to clarify, if we consider an array with 2 dimension. Say the 1st one didn't go through the happy path, so we log the error, update an error flag and continue. Then say if the 2nd dimension goes through the happy path, should the `resultType`  be set to an error type regardless or should the below code be executed for that loop? 
```
arrayTypeSymbol.type = arrType;
resultType = arrayTypeSymbol.type;
markParameterizedType(arrType, arrType.eType);
```